subsys/usb/device/class/cdc_acm.c

```diff
@@ -56,6 +56,11 @@
 /* definitions */
 
 #include <zephyr/logging/log.h>
+#if CONFIG_USB_CDC_ACM_LOG_LEVEL != 0
+/* Sanity check for endless recursive logging loop */
+BUILD_ASSERT(!DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_console), zephyr_cdc_acm_uart)),
+	     "CDC ACM logging must be disabled when used as console");
+#endif
```

Comment:
I am afraid it will cause some trouble, for example https://github.com/zephyrproject-rtos/zephyr/pull/52293/commits/934c722c33d659c825a7b175e57d6695cf1ddba4#diff-919f251088f0f549acf6c48fe86aad3bdaffcb1cdfb8193c7cc59bd6a0ee5718R116-R118

Maybe `#if CONFIG_USB_CDC_ACM_LOG_LEVEL > LOG_LEVEL_ERR` ?

Reply:
Well, `LOG_LEVEL_ERR` can lead to endless recursive loop due to the `LOG_ERR` call at `cdc_acm_poll_out`:
https://github.com/zephyrproject-rtos/zephyr/blob/fed6bde788522fbbacf7196280c700f0be3feaf1/subsys/usb/device/class/cdc_acm.c#L1014

So maybe just change the `USB_CDC_ACM_LOG_LEVEL` to 0 if board uses CDC ACM as console. If the CDC ACM itself uses logging when it is the logger output, then we'll always be prone to issues. I think the `LOG_ERR` is appropriate level (i.e. it should not be lowered) in `cdc_acm_poll_out` as the action directly violates UART poll API.

Reply:
>So maybe just change the USB_CDC_ACM_LOG_LEVEL to 0 if board uses CDC ACM as console. If the CDC ACM itself uses logging when it is the logger output, then we'll always be prone to issues. I think the LOG_ERR is appropriate level (i.e. it should not be lowered) in cdc_acm_poll_out as the action directly violates UART poll API.

You are right, my only concern is that someone's application will stop building after this, and since many do not take the time to deal with configuration, we will probably see a few new bug reports for a short time.
Do you like to insert a note to https://docs.zephyrproject.org/latest/services/usb/uds_cdc_acm.html (_doc/services/usb/uds_cdc_acm.rst_) regarding USB_CDC_ACM_LOG_LEVEL?

Reply:
If we want to retain the possibility for applications to build, then we can warn about incorrect log level and force it to none if CDC ACM is used as console.

Reply:
Like
```c
#if DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_console), zephyr_cdc_acm_uart))
#warning "USB_CDC_ACM_LOG_LEVEL forced to LOG_LEVEL_NONE"
#define CDC_ACM_LOG_LEVEL LOG_LEVEL_NONE
#elif
#define CDC_ACM_LOG_LEVEL CONFIG_USB_CDC_ACM_LOG_LEVEL
#endif
LOG_MODULE_REGISTER(usb_cdc_acm, CDC_ACM_LOG_LEVEL);
```
?

Reply:
Yes, but with `#undef CDC_ACM_LOG_LEVEL` before redefining it.

Reply:
okay